fs/copy.py

```diff
@@ -56,13 +88,35 @@ def copy_file(src_fs, src_path, dst_fs, dst_path):
             if src_fs is dst_fs:
                 # Same filesystem, so we can do a potentially optimized copy
                 src_fs.copy(src_path, dst_path, overwrite=True)
+                return dst_path
```

Comment:
Should we catch the copy failed event? Something like:
def copy_file(src_fs, src_path, dst_fs, dst_path):
try:
    with manage_fs(src_fs, writeable=False) as src_fs:
     ....
     return True
except:
     return False

We can return a boolean to report on the success of the copy.
Again here we should use more detailed Exception handling.
What do you think?